src/main/java/ch/njol/skript/expressions/ExprChatRecipients.java

```diff
@@ -40,71 +40,68 @@
 import ch.njol.util.coll.CollectionUtils;
 
 @Name("Chat Recipients")
-@Description("Recipients of chat events where this is called.")
+@Description("Recipients of chat/broadcast events where this is called.")
 @Examples("chat recipients")
-@Since("2.2-Fixes-v7, 2.2-dev35 (clearing recipients)")
-public class ExprChatRecipients extends SimpleExpression<Player> {
+@Since("2.2-Fixes-v7, 2.2-dev35 (clearing recipients), INSERT VERSION (broadcast event)")
+public class ExprChatRecipients extends SimpleExpression<CommandSender> {
 
 	static {
-		Skript.registerExpression(ExprChatRecipients.class, Player.class, ExpressionType.SIMPLE, "[chat][( |-)]recipients");
-	}
-
-	@Override
-	public boolean isSingle() {
-		return false;
-	}
-
-	@Override
-	public Class<Player> getReturnType() {
-		return Player.class;
-	}
-
-	@Override
-	public Class<?>[] acceptChange(final ChangeMode mode) {
-		return CollectionUtils.array(Player[].class);
+		Skript.registerExpression(ExprChatRecipients.class, CommandSender.class, ExpressionType.SIMPLE, "[chat][( |-)]recipients");
 	}
 
 	@Override
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
-		if (!(getParser().isCurrentEvent(AsyncPlayerChatEvent.class))) {
-			Skript.error("Cannot use chat recipients expression outside of a chat event", ErrorQuality.SEMANTIC_ERROR);
+		if (!(getParser().isCurrentEvent(AsyncPlayerChatEvent.class, BroadcastMessageEvent.class))) {
+			Skript.error("Cannot use chat recipients expression outside of a chat or server broadcast event");
 			return false;
 		}
 		return true;
 	}
 
 	@Override
-	public String toString(@Nullable Event event, boolean debug) {
-		return "chat recipients";
+	@Nullable
+	protected CommandSender[] get(Event event) {
+		if (event instanceof AsyncPlayerChatEvent) {
+			return ((AsyncPlayerChatEvent) event).getRecipients().toArray(new Player[0]);
+		} else if (event instanceof BroadcastMessageEvent) {
+			return ((BroadcastMessageEvent) event).getRecipients().toArray(new CommandSender[0]);
+		}
+		return null;
 	}
 
 	@Override
-	@Nullable
-	protected Player[] get(Event event) {
-		if (!(event instanceof AsyncPlayerChatEvent))
-			return null;
-
-		AsyncPlayerChatEvent ae = (AsyncPlayerChatEvent) event;
-		Set<Player> playerSet = ae.getRecipients();
-		return playerSet.toArray(new Player[playerSet.size()]);
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		return CollectionUtils.array(CommandSender[].class);
 	}
 
 	@Override
 	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
-		if (!(event instanceof AsyncPlayerChatEvent))
+		if (!(event instanceof AsyncPlayerChatEvent) || !(event instanceof BroadcastMessageEvent))
```

Comment:
Did you even test this? This condition will never pass. An event can't be an instance of both of those at the same time.